### PR TITLE
WP-52: File Structure and Simple Navigation Bar

### PR DESCRIPTION
**Pull request by ltan02**
**Description:**
Setup a basic file structure to organize the frontend. Implemented a simple navigation bar that will be used to switch between the main pages. 

**Changes:**
- Added basic folders including: `routes` (used for switching between pages), `ui` (used for the design of the different pages), and `widgets`.
- Added a simple navigation bar that can be used to switch between the `dashboard_page`, `my_workouts_page`, and `creators_corner_page`.

image

**Related Tickets:**
[https://ubclaunchpad.atlassian.net/browse/WP-52](url)

### CODE REVIEW

File: lib/app_view.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+import 'package:workoutpersonalizer_frontend/widgets/navigation_bar/navigation_bar.dart';
```
Comment (richardyjtian) [inline lib/app_view.dart]: we should note somewhere that we're using absolute paths so we can keep consistent with everyone

File: lib/ui/creators_corner_page.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+class CreatorsCornerPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Container(
+        child: const Text(
+          "Creator's Corner Page",
```
Comment (richardyjtian) [inline lib/ui/creators_corner_page.dart:10]: let's add a TODO comment here linking WP-13

File: lib/ui/dashboard_page.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+class DashboardPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Container(
+        child: const Text(
+          "Dashboard Page",
```
Comment (richardyjtian) [inline lib/ui/dashboard_page.dart]: same as above, though I don't think we have a ticket for the dashboard yet

File: lib/ui/my_workouts_page.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+class MyWorkoutsPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Container(
+        child: const Text(
+          "My Workouts Page",
```
Comment (richardyjtian) [inline lib/ui/my_workouts_page.dart:9]: I can add the TODO for this section after I merge my changes for this section

File: lib/widgets/navigation_bar/interactive_nav_item.dart [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:universal_html/html.dart' as html;
+
+class InteractiveNavItem extends MouseRegion {
+  static final appContainer =
```
Comment (richardyjtian) [inline lib/widgets/navigation_bar/interactive_nav_item.dart]: is there maybe a better way to change the mouse pointer on hover? if not, then this is okay

**Review by richardyjtian — DISMISSED** [review]
Great work Lance! Just some minor comments you can get to in either this PR or the next

File: lib/app_view.dart (hunk 1 above)
Comment (ltan02) [reply lib/app_view.dart]: I added a note in the `main.dart` file, which felt like the best place to put it.

File: lib/ui/creators_corner_page.dart (hunk 2 above)
Comment (ltan02) [reply lib/ui/creators_corner_page.dart:10]: I added a TODO here, can you check to see if this is what you wanted me to put in the TODO?

File: lib/ui/dashboard_page.dart (hunk 3 above)
Comment (ltan02) [reply lib/ui/dashboard_page.dart]: Added the dashboard ticket to the TODO.

File: lib/widgets/navigation_bar/interactive_nav_item.dart (hunk 5 above)
Comment (ltan02) [reply lib/widgets/navigation_bar/interactive_nav_item.dart]: I simplified the mouse pointer on hover by using the built-in `cursor` property of `MouseRegion`, which made the `interactive_nav_item.dart` file redundant and was removed.

File: lib/ui/my_workouts_page.dart (hunk 4 above)
Comment (ltan02) [reply lib/ui/my_workouts_page.dart:9]: Sure, I won't add a TODO for this.

File: lib/main.dart [hunk 6]
```diff
@@ -1,115 +1,31 @@
+// Absolute pathing is used throughout this project
```
Comment (lilyydu) [inline lib/main.dart:1]: We can move this to the ContributingMD instead so we can delete it here & I can make the ticket for it

File: lib/routes/routes_generator.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:workoutpersonalizer_frontend/routes/routes.dart';
+import 'package:workoutpersonalizer_frontend/ui/creators_corner_page.dart';
+import 'package:workoutpersonalizer_frontend/ui/dashboard_page.dart';
+import 'package:workoutpersonalizer_frontend/ui/my_workouts_page.dart';
+
+class RouteGenerator {
+  static Route<dynamic> generateRoute(RouteSettings settings) {
+    switch (settings.name) {
+      case routeDashboard:
+        return MaterialPageRoute(builder: (_) => DashboardPage());
```
Comment (lilyydu) [inline lib/routes/routes_generator.dart]: Just to remove redundancy but maybe we can set the page in the switch and only call 
`return MaterialPageRoute(builder: (_) =>` once?

File: lib/routes/routes.dart [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+
+const String routeDashboard = '/dashboard';
```
Comment (lilyydu) [inline lib/routes/routes.dart]: More of a grammatical thing but possibly dashboardRoute, myWorkoutsRoute and creatorsCornerRoute instead? Last two are a bit wordy but we can figure out alternatives later :)

File: lib/ui/creators_corner_page.dart [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import 'package:flutter/material.dart';
```
Comment (lilyydu) [inline lib/ui/creators_corner_page.dart:1]: Later on for us to keep in mind but if we import more duplicate packages for all the pages, might be nice to define a parent file & just reference that!

File: lib/ui/creators_corner_page.dart [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+class CreatorsCornerPage extends StatelessWidget {
+  @override
```
Comment (lilyydu) [inline lib/ui/creators_corner_page.dart:4]: We'll need to add in linting later on, do you mind adding a ticket for this? :)

File: lib/ui/creators_corner_page.dart [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+  Widget build(BuildContext context) {
+    // TODO: WP-13 (ubclaunchpad.atlassian.net/browse/WP-13)
+    return Center(
+      child: Container(
+        child: const Text(
+          "Creator's Corner Page",
+          style: TextStyle(fontSize: 30),
```
Comment (lilyydu) [inline lib/ui/creators_corner_page.dart:11]: Might be good to have the styling details as constants in a separate file for reusability & ease of change!

File: lib/routes/routes_generator.dart [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+      case routeCreatorsCorner:
+        return MaterialPageRoute(builder: (_) => CreatorsCornerPage());
```
Comment (lilyydu) [inline lib/routes/routes_generator.dart]: Also not sure if this is just me but I'm seeing duplicates on DevTools!

[Screen Shot 2021-11-11 at 5 44 28 PM](https://user-images.githubusercontent.com/54044854/141394455-f76a93c7-d41e-4f51-93ea-820707d110a8.png)

**Review by lilyydu — CHANGES_REQUESTED** [review]
Awesome job Lance!  🎉    Great work taking initiative with the structure  🙌    Just left a couple of small comments :)

File: lib/routes/routes_generator.dart (hunk 12 above)
Comment (richardyjtian) [reply lib/routes/routes_generator.dart]: Yeah everytime I click on a nav bar item, it seems to be creating a new page and adding it to expanded. Good catch Lily

**Review by richardyjtian — APPROVED** [review]

**Review by lilyydu — APPROVED** [review]
Thanks for making the changes! Great work :)
